c_glib/gandiva-glib/native-function.cpp

```diff
@@ -0,0 +1,209 @@
+    break;
+  default:
+    G_OBJECT_WARN_INVALID_PROPERTY_ID(object, prop_id, pspec);
+    break;
+  }
+}
+
+static void
+ggandiva_native_function_init(GGandivaNativeFunction *object)
+{
+}
+
+static void
+ggandiva_native_function_class_init(GGandivaNativeFunctionClass *klass)
+{
+  auto gobject_class = G_OBJECT_CLASS(klass);
+
+  gobject_class->dispose = ggandiva_native_function_dispose;
+  gobject_class->finalize = ggandiva_native_function_finalize;
+  gobject_class->set_property = ggandiva_native_function_set_property;
+
+  GParamSpec *spec;
+  spec = g_param_spec_pointer("native_function",
+                              "NativeFunction",
+                              "The raw gandiva::NativeFunction *",
+                              static_cast<GParamFlags>(G_PARAM_WRITABLE |
+                                                       G_PARAM_CONSTRUCT_ONLY));
+  g_object_class_install_property(gobject_class, PROP_NATIVE_FUNCTION, spec);
+}
+
+/**
+ * ggandiva_native_function_get_function_signature:
+ * @native_function: A #GGandivaNativeFunction.
+ *
+ * Returns: (transfer full): A #GGandivaFunctionSignature that represents
+ *   the signature of the native function.
+ */
+GGandivaFunctionSignature *
+ggandiva_native_function_get_function_signature(GGandivaNativeFunction *native_function)
+{
+  auto priv = GGANDIVA_NATIVE_FUNCTION_GET_PRIVATE(native_function);
+  auto gandiva_function_signature = priv->native_function->signature();
+  return ggandiva_function_signature_new_raw(&gandiva_function_signature);
+}
+
+/**
+ * ggandiva_native_function_get_string_function_signature:
+ * @native_function: A #GGandivaNativeFunction.
+ *
+ * Returns: (transfer full):
+ *   The string representation of the signature of the native function.
+ */
+gchar *
+ggandiva_native_function_get_string_function_signature(GGandivaNativeFunction *native_function)
+{
+  auto priv = GGANDIVA_NATIVE_FUNCTION_GET_PRIVATE(native_function);
+  auto gandiva_function_signature = priv->native_function->signature();
+  return g_strdup(gandiva_function_signature.ToString().c_str());
+}
+
+/**
+ * ggandiva_native_function_get_result_nullable_type:
+ * @native_function: A #GGandivaNativeFunction.
+ *
+ * Returns:
+ *   A value of #GGandivaResultNullableType.
+ */
+GGandivaResultNullableType
+ggandiva_native_function_get_result_nullable_type(GGandivaNativeFunction *native_function)
+{
+  auto priv = GGANDIVA_NATIVE_FUNCTION_GET_PRIVATE(native_function);
+  auto gandiva_result_nullable_type = priv->native_function->result_nullable_type();
+  return static_cast<GGandivaResultNullableType>(gandiva_result_nullable_type);
+}
+
+/**
+ * ggandiva_native_function_needs_context:
+ * @native_function: A #GGandivaNativeFunction.
+ *
+ * Returns:
+ *   %TRUE if the native function needs a context for evaluation,
+ *   %FALSE otherwise.
+ */
+gboolean
+ggandiva_native_function_needs_context(GGandivaNativeFunction *native_function)
```

Comment:
Could you use `need` instead of `needs` because our other API doesn't use `s` like `gets`?